File: frontend/Assets/Scripts/DataTransfer/Server/TurnDataSCmd.cs
using System;
using System.IO;
using Attributes;
using DataTransfer.Data;
using Utils;


namespace DataTransfer.Server {

    [DTO (DTOCode.TurnDataServer)]
    public class TurnDataSCmd : ServerCommand {

        public static event Action <TurnDataSCmd> OnReceived;

        public TurnData Data { get; private set; }


        public override void ReadMembers (BinaryReader reader) {
            Data = new TurnData ();
            Data.ReadMembers (reader);
        }


        public override void Execute () {
            OnReceived._ (this);
        }

    }

}

File: backend/src/main/java/dto/server/TurnDataSCmd.java
package dto.server;

import dto.DTOCode;
import dto.DTOs;
import dto.data.TurnData;
import io.netty.buffer.ByteBuf;


@DTOCode(DTOs.TURN_DATA_SERVER)
public class TurnDataSCmd extends ServerCommand {

    private final TurnData data;


    public TurnDataSCmd (TurnData data) {
        this.data = data;
    }


    @Override
    public void writeMembers (ByteBuf byteBuf) {
        data.writeMembers(byteBuf);
    }
}
